src/components/Staking/WithdrawalCredentials.tsx

```diff
@@ -0,0 +1,155 @@
+    () => (validator ? `0x${validator.withdrawalCredentials.slice(-40)}` : ""),
+    [validator]
+  )
+  const shortAddress = useMemo<string>(
+    () =>
+      longAddress ? `${longAddress.slice(0, 6)}…${longAddress.slice(-4)}` : "",
+    [longAddress]
+  )
+  const resultText = useMemo<string | JSX.Element>(() => {
+    if (isLoading) return <Spinner />
+    if (hasError)
+      return (
+        <Flex bg="errorNeutral" p={4}>
+          <Text m={0} color="error">
+            Oops! Double check validator index number and try again.
+          </Text>
+        </Flex>
+      )
+    if (!validator) return " "
+    if (validator.isUpgraded)
+      return (
+        <Flex bg="successNeutral" p={4}>
+          <Text m={0} color="success">
+            <Text as="span" fontWeight="bold">
+              Validator index {validator.validatorIndex} is ready to start
+              receiving rewards!
+            </Text>{" "}
+            Withdrawal credentials linked to execution address{" "}
+            <CopyToClipboard text={longAddress} inline>
+              {(isCopied) => (
+                <>
+                  <Text as="span" title={longAddress} fontWeight="bold">
+                    {shortAddress}
+                  </Text>
+                  {isCopied ? (
+                    <>
+                      <Emoji text="✅" fontSize="lg" mr={2} ml={2} />
+                      <Text as="span" title={longAddress}>
+                        Copied!
+                      </Text>
+                    </>
+                  ) : (
+                    <Emoji text="📋" fontSize="lg" mr={2} ml={2} />
+                  )}
+                </>
+              )}
+            </CopyToClipboard>
+          </Text>
+        </Flex>
+      )
+    return (
+      <Flex bg="errorNeutral" p={4}>
+        <Text m={0} color="error">
+          <Text as="span" fontWeight="bold">
+            This {validator.isTestnet ? "Goerli testnet" : ""} validator needs
+            to be upgraded.
+          </Text>{" "}
+          Instructions on how to upgrade can currently be found at{" "}
+          <Link to="https://zhejiang.launchpad.ethereum.org/withdrawals">
+            Zhejiang Testnet Staking Launchpad
+          </Link>
+        </Text>
+      </Flex>
+    )
+  }, [isLoading, hasError, validator, longAddress, shortAddress])
+
+  return (
+    <Flex direction="column" gap={4}>
+      <Flex alignItems="center" gap={2} flexWrap="wrap">
+        <Input
+          size="lg"
+          id="validatorIndex"
+          value={inputValue}
+          onChange={handleChange}
+          w={{ base: "full", sm: "18ch" }}
+          placeholder="Validator index"
+          bg="background"
+        />
+        <Flex
+          w={{ base: "full", sm: "fit-content" }}
+          direction={{ base: "column", sm: "row" }}
+          gap={2}
+        >
+          <Button
+            onClick={() => checkWithdrawalCredentials()}
+            disabled={!inputValue.length}
```

Comment:
Since we have an `isLoading` flag, perhaps we could use this pattern https://chakra-ui.com/docs/components/button#button-loading-state instead of disabling the button.

`<Button isLoading={isLoading}>`

Reply:
@pettinarip Not sure I understand... The disabled state is when no number is entered, not for when it's loading. Or are you referring to replacing the existing loading spinner with a loader inside the button?

Reply:
I've removed the spinner below the input and replaced with `isLoading` on the button... I think this looked better. Kept the disabled state though when no number inputted.

Reply:
Oh got it. Yeah, I thought you were disabling it when it was loading. Didn't realize that you had that other behavior.